### PR TITLE
In-code documentation for files.py

### PR DESCRIPTION
**Pull request by therealsilence**
- Formatting for readability
- Add 'doc' strings for types, forms, props.

### CODE REVIEW

File: synapse/models/files.py [hunk 1]
```diff
@@ -121,62 +121,92 @@ def seedFileSha1(self, prop, valu, **props):
...
 
                 ('file:txtref', {}, [
-                    ('file', {'ptype': 'file:bytes', 'ro': 1}),
-                    ('xref', {'ptype': 'propvalu', 'ro': 1}),
-                    ('xref:prop', {'ptype': 'str', 'ro': 1}),
-                    ('xref:intval', {'ptype': 'int', 'ro': 1}),
-                    ('xref:strval', {'ptype': 'str', 'ro': 1}),
+                    ('file', {'ptype': 'file:bytes', 'doc': 'The guid of the file containing the reference.', 'ro': 1}),
+                    ('xref', {'ptype': 'propvalu',
+                         'doc': 'The form=valu of the object referenced in the fiole, e.g., inet:fqdn=foo.com.', 'ro': 1}),
```
Comment (vertexmc) [inline synapse/models/files.py]: file

File: synapse/models/files.py [hunk 2]
```diff
@@ -121,62 +121,92 @@ def seedFileSha1(self, prop, valu, **props):
...
+                         'doc': 'The value of the property of the referenced object, as specified by the propvalu, if the value is a string.', 'ro': 1}),
                 ]),
 
                 ('file:path', {}, (
-                    ('dir', {'ptype': 'file:path', 'doc': 'The parent directory for this path.'}),
-                    ('ext', {'ptype': 'str:lwr', 'doc': 'The file extension ( if present ).'}),
-                    ('base',
-                     {'ptype': 'file:base', 'doc': 'The final path component, such as the filename, of this path.'}),
+                    ('dir', {'ptype': 'file:path',
+                         'doc': 'The parent directory of the file path. Can be the directory path (if the file:path ends in a file name) or the parent directory (if the file:path is itself a directory).', 'ro': 1}),
```
Comment (vertexmc) [inline synapse/models/files.py]: This doc is confusing to me. A `file:path` is a path that can be to either a file or a directory.

File: synapse/models/files.py [hunk 3]
```diff
@@ -121,62 +121,92 @@ def seedFileSha1(self, prop, valu, **props):
...
+                    ('ext', {'ptype': 'str:lwr',
+                         'doc': 'The file extension of the file name, (if present); for example: exe, bat, py, docx.', 'ro': 1}),
+                    ('base', {'ptype': 'file:base',
+                          'doc': 'The final component of the file path. Can be a file name (if present) or the final directory.', 'ro': 1}),
                 )),
 
-                ('file:base', {'ptype': 'file:base', 'doc': 'A final path component, such as the filename.'}, ()),
+                ('file:base', {'ptype': 'file:base'}, (
```
Comment (vertexmc) [inline synapse/models/files.py]: Doc was removed because the type is already doc'd.

File: synapse/models/files.py [hunk 4]
```diff
@@ -121,62 +121,92 @@ def seedFileSha1(self, prop, valu, **props):
...
-                    ('sha512', {'ptype': 'hash:sha512'}),
-                    ('name', {'ptype': 'file:base', 'doc': 'For display purposes only'}),
-                    ('mime', {'ptype': 'str', 'defval': '??', 'doc': 'Mime type for the file bytes'}),
+                    ('size', {'ptype': 'int', 'doc': 'The size of the file in bytes.', 'ro': 1}),
+                    ('md5', {'ptype': 'hash:md5', 'doc': 'The md5 hash of the file.', 'ro': 1}),
+                    ('sha1', {'ptype': 'hash:sha1', 'doc': 'The sha1 hash of the file.', 'ro': 1}),
+                    ('sha256', {'ptype': 'hash:sha256', 'doc': 'The sha256 hash of the file.', 'ro': 1}),
+                    ('sha512', {'ptype': 'hash:sha512', 'doc': 'The sha512 hash of the file.', 'ro': 1}),
+                    ('name', {'ptype': 'file:base',
+                          'doc': 'The name of the file. Because a given set of bytes can have any number of arbitrary names, this field is used for display purposes only.'}),
```
Comment (vertexmc) [inline synapse/models/files.py]: I like this explanation

File: synapse/models/files.py (hunk 1 above)
Comment (therealsilence) [reply synapse/models/files.py]: Fixed

File: synapse/models/files.py (hunk 2 above)
Comment (therealsilence) [reply synapse/models/files.py]: Changed - removed all but first sentence. Over-explaining.

**Pull request merged**